server/routes/sockets/room-service.js

```diff
@@ -130,12 +170,24 @@ class RoomService {
             return;
         }
         try {
+            // First we need to check if the user is authorized for this action.
+            if (!await this.checkIfUserAuthorized(
+                eventInfo.authToken, socket,
+                {
+                    "title": "Failed to Create Room",
+                    "message": "We could not authorize your attempt to create a room."
+                }
+            )) {
+                return;
+            }
```

Comment:
Now that I see how `checkIfUserAuthorized` is used, it feel like it shouldn't bear the responsibility of sending a response, not only because its name doesn't seem to indicate it does. This big if conditional looks messy to me. I'd leave `checkIfUserAuthorized` as a simpler function that just returns true or false, then handle sending an error event in this function, since the message is already here:
```js
const userIsAuthorized = await this.checkIfUserAuthorized(eventInfo.authToken, socket);
if (!userIsAuthorized) {
    return this.emitUserEvent('not_authorized', socket, {
        "title": "Failed to Create Room",
        "message": "We could not authorize your attempt to create a room."
    });
}
```